packages/loader/container-loader/src/connectionStateHandler.ts

```diff
@@ -29,6 +29,18 @@ export interface IConnectionStateHandlerInputs {
     connectionStateChanged: () => void;
 }
 
+/** 
+ * A type that contains details pertaining to the connection state 
+ */
+export type ConnectionStateDetails =
+    | {
+        state: ConnectionState.Connected | ConnectionState.CatchingUp;
+    }
+    | {
+        state: ConnectionState.Disconnected;
+        reason?: string;
```

Comment:
In general, I find it useful to have that data at different layers.
But, this is an API contract! Meaning that we might be running into various back-compat concerns here (like updating socket.io version may result in different strings, and code breaking in production).
So if you want to add it here, we need a very strong comment here saying that this string can be used only to be logged to telemetry, and that specific values will change over time - none of shipping code can take a dependency on actual values other than logging to telemetry.

Reply:
+1

Please do include some documentation explaining that this is for diagnostic/telemetry purposes only. Any exported API with this parameter should have doc comments explaining this, and the README should be updated to note that the disconnected event may have a string argument which is this reason, to be used only for diagnostic purposes.